File: field-manager-dotnet-backend/Controllers/FieldMonitorController.cs
using FieldManagerDotnetBackend.Models;
using FieldManagerDotnetBackend.Services;
using Microsoft.AspNetCore.Mvc;
using System;

namespace FieldManagerDotnetBackend.Controllers
{
    /// <summary>
    /// Provides endpoints for retrieving field zone information such as
    /// biomass indices, weather forecasts and soil properties. This
    /// corresponds to the "Field Monitor" feature in xarvio FIELD
    /// MANAGER.
    /// </summary>
    [ApiController]
    [Route("api/fields/{fieldId:guid}/monitor")]
    public class FieldMonitorController : ControllerBase
    {
        private readonly DemoDataService _dataService;

        public FieldMonitorController(DemoDataService dataService)
        {
            _dataService = dataService;
        }

        /// <summary>
        /// Retrieves zone data for a specified field. Returns 404 if no
        /// matching field exists.
        /// </summary>
        /// <param name="fieldId">Identifier of the field</param>
        /// <returns>ZoneData representing biomass and weather information</returns>
        [HttpGet]
        public ActionResult<ZoneData> GetFieldMonitor(Guid fieldId)
        {
            var field = _dataService.GetField(fieldId);
            if (field is null || field.ZoneData is null)
            {
                return NotFound();
            }
            return Ok(field.ZoneData);
        }
    }
}

File: fieldmanager-spring-backend/src/main/java/com/xarvio/fieldmanager/controllers/FieldMonitorController.java
package com.xarvio.fieldmanager.controllers;

import java.util.UUID;
import lombok.RequiredArgsConstructor;

import com.xarvio.fieldmanager.models.Field;
import com.xarvio.fieldmanager.models.ZoneData;
import com.xarvio.fieldmanager.services.DemoDataService;

import org.springframework.web.bind.annotation.RestController;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@RestController
@RequiredArgsConstructor
public class FieldMonitorController {

    private final DemoDataService demoDataService;

    @GetMapping("api/fields/{fieldId}/monitor")
    public ResponseEntity<ZoneData> getFieldMonitor(@PathVariable UUID fieldId) {
        return demoDataService.getField(fieldId)
                .map(Field::zoneData)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

}
